Bottom: 1238998992b7973f475d4d4959f419d7f03aa417
Top:    7220a4f68d3fa9ef8cd5c061f67e241f919c7bdf
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-21 19:00:12 +0000

Refresh of try-something-with-the-rtrees

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 2d18e08..43e167d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2388,7 +2388,6 @@ poly_FreeContours (PLINE **pline)
       *pline = (*pl)->next;
       poly_DelContour (pl);
     }
-  r_destroy_tree (&P->contour_tree);
 }
 
 void
@@ -2401,11 +2400,13 @@ poly_Free (POLYAREA ** p)
   for (cur = (*p)->f; cur != *p; cur = (*p)->f)
     {
       poly_FreeContours (&cur->contours);
+      r_destroy_tree (&cur->contour_tree);
       cur->f->b = cur->b;
       cur->b->f = cur->f;
       free (cur);
     }
     poly_FreeContours (&cur->contours);
+    r_destroy_tree (&cur->contour_tree);
   free (*p), *p = NULL;
 }
